tests/iterators.rs

```diff
@@ -564,6 +564,65 @@ fn test_fold() {
     assert_eq!(a.iter().fold(0, |acc, &x| acc + x), 1);
 }
 
+#[test]
```

Comment:
For the sake of readability, would it make sense to split these checks in multiple test cases?

Reply:
Yeah, I agree. I've split it up.